src/components/map/Map.js

```diff
@@ -31,107 +22,83 @@ const layerType = {
 };
 
 const styles = {
-    mapContainer: {
+    root: {
         height: '100%',
-        '& img': {
-            // Override from ui/core/UI'
-            maxWidth: 'none',
-        },
-        '& .leaflet-control-zoom a': {
-            // Override from ui/core/UI'
-            color: 'black!important',
-        },
-        '& .leaflet-control-measure a': {
-            fontSize: '12px!important',
-        },
-        '& .leaflet-popup-content th': {
-            fontWeight: 'bold',
-            paddingRight: 5,
-        },
-    },
-    mapDownload: {
-        // Roboto font is not loaded by dom-to-image => switch to Arial
-        '& div': {
-            fontFamily: 'Arial,sans-serif!important',
-        },
-        '& .leaflet-control-zoom, & .leaflet-control-geocoder, & .leaflet-control-measure, & .leaflet-control-fit-bounds': {
-            display: 'none!important',
-        },
     },
 };
 
 class Map extends Component {
-    static contextTypes = {
-        map: PropTypes.object,
-    };
-
     static propTypes = {
+        isPlugin: PropTypes.bool,
         basemap: PropTypes.object,
-        basemaps: PropTypes.array,
+        layers: PropTypes.array,
+        controls: PropTypes.array,
         bounds: PropTypes.array,
-        dataTableOpen: PropTypes.bool,
-        dataTableHeight: PropTypes.number,
-        isDownload: PropTypes.bool,
-        interpretationsPanelOpen: PropTypes.bool,
-        layersPanelOpen: PropTypes.bool,
-        legendPosition: PropTypes.string,
         latitude: PropTypes.number,
         longitude: PropTypes.number,
-        mapViews: PropTypes.array,
-        showName: PropTypes.bool,
         zoom: PropTypes.number,
         coordinatePopup: PropTypes.array,
-        closeCoordinatePopup: PropTypes.func.isRequired,
+        closeCoordinatePopup: PropTypes.func,
         openContextMenu: PropTypes.func.isRequired,
+        onCloseContextMenu: PropTypes.func,
         classes: PropTypes.object.isRequired,
     };
 
-    componentWillMount() {
-        this.context.map.on('contextmenu', this.onRightClick, this);
-    }
+    static defaultProps = {
+        isPlugin: false,
+    };
 
-    componentDidMount() {
-        const { bounds, latitude, longitude, zoom } = this.props;
-        const map = this.context.map;
+    static childContextTypes = {
+        map: PropTypes.object.isRequired,
+    };
 
-        this.node.appendChild(map.getContainer()); // Append map container to DOM
+    getChildContext() {
```

Comment:
Yes it could, but I'm moving away from the MapProvider, as it don't work with split view which requires several map instances. The reason MapProvider is still is used is a simple check if a certain map layer is supported. My plan is to remove the MapProvider in a later PR.

Reply:
I removed the MapProvider in this PR - it is obsolete.